src/BehaviorsSDKManaged/Microsoft.Xaml.Interactions/Core/ChangePropertyAction.cs

```diff
@@ -120,7 +122,9 @@ public object Execute(object sender, object parameter)
         private void UpdatePropertyValue(object targetObject)
         {
             Type targetType = targetObject.GetType();
-            PropertyInfo propertyInfo = targetType.GetRuntimeProperty(this.PropertyName.Path);
+            var properties = targetType.GetRuntimeProperties()
```

Comment:
This may cause some trouble as the `GetRuntimeProperties` may return properties out of order. There's a need to lookup  for the property by descending from `targetType`, through `BaseType` to `System.Object`.

Reply:
Do you want me to filter properties declaring type by TextBlock?

Reply:
Do you mean retrieve all properties, then sort through all the hierarchy and take last?

Reply:
Yes, a simple while loop that iterates through the hierarchy using BaseType would be enough.

Reply:
@jeromelaban done